config/consensus.go

```diff
@@ -1147,36 +1147,66 @@ func initConsensusProtocols() {
 	// v31 can be upgraded to v32, with an update delay of 7 days ( see calculation above )
 	v31.ApprovedUpgrades[protocol.ConsensusV32] = 140000
 
-	// ConsensusFuture is used to test features that are implemented
-	// but not yet released in a production protocol version.
-	vFuture := v32
-	vFuture.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+	v33 := v32
+	v33.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
 
 	// Make the accounts snapshot for round X at X-CatchpointLookback
-	vFuture.CatchpointLookback = 320
+	// order to guarantee all nodes produce catchpoint at the same round.
+	v33.CatchpointLookback = 320
 
 	// Require MaxTxnLife + X blocks and headers preserved by a node
-	vFuture.DeeperBlockHeaderHistory = 1
+	v33.DeeperBlockHeaderHistory = 1
+
+	v33.MaxTxnBytesPerBlock = 5 * 1024 * 1024
+
+	// Set upgrade values to some small values to allow v33 -> v34 fast upgrade
+	v33.UpgradeVoteRounds = 100
+	v33.MinUpgradeWaitRounds = 100
```

Comment:
would it be easier if we put 10000 rounds between v33 and v34?

Reply:
I think we need at least 1560, if I understand where the state proof issue is headed.

Reply:
this is temporary on v33 before stateproof that is enabled in v34

Reply:
Right, but I think that when stateproof is enabled, it will need 2560 of history.  It's the same issue as the 1001 of history that's causing the double upgrade for block header access in AVM.

Reply:
@algoidan is the genesis of SP handled as a special case for block history?

Reply:
When state proof chain starts it will not require longer history. so I don't see any problems here. (We would verify that in a manual test)

just to be clear, state proofs would require a long history of blocks only if it is lagging behind. definitely not at startup

Reply:
put 10000 and tested